pkg/integration/prometheus/config_analyzer.go

```diff
@@ -0,0 +1,287 @@
+			for _, c := range pod.Spec.Containers {
+				if c.Name == prometheusContainerName {
+					proms = append(proms, pod)
+				}
+			}
+		}
+	}
+	return proms, nil
+}
+
+func findPrometheusConfigPath(ctx context.Context, client kubernetes.Interface, pod *corev1.Pod) (string, error) {
+	var path string
+	var err error
+	for _, container := range pod.Spec.Containers {
+		for _, arg := range container.Args {
+			if strings.HasPrefix(arg, prometheusConfigFlag) {
+				path = strings.TrimLeft(arg, prometheusConfigFlag)
+			}
+			if strings.HasPrefix(arg, configReloaderConfigFlag) {
+				path = strings.TrimLeft(arg, configReloaderConfigFlag)
+			}
+		}
+		if container.Name == configReloaderContainerName {
+			return path, nil
+		}
+	}
+	if path == "" {
+		err = fmt.Errorf("prometheus config path not found in pod: %s", pod.Name)
+	}
+	return path, err
+}
+
+func findPrometheusConfigVolumeAndKey(ctx context.Context, client kubernetes.Interface, pod *corev1.Pod) (*corev1.Volume, string, error) {
+	path, err := findPrometheusConfigPath(ctx, client, pod)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Find the volumeMount the config path is pointing to.
+	var volumeName = ""
+	for _, container := range pod.Spec.Containers {
+		for _, vm := range container.VolumeMounts {
+			if strings.HasPrefix(path, vm.MountPath) {
+				volumeName = vm.Name
+				break
+			}
+		}
+	}
+
+	// Get the actual Volume from the name.
+	for _, volume := range pod.Spec.Volumes {
+		if volume.Name == volumeName {
+			return &volume, filepath.Base(path), nil
+		}
+	}
+
+	return nil, "", errors.New("volume for Prometheus config not found")
+}
+
+func extractPrometheusConfigFromVolume(ctx context.Context, client kubernetes.Interface, volume *corev1.Volume, namespace, key string) ([]byte, error) {
+	var b []byte
+	var ok bool
+	// Check for Secret volume.
+	if vs := volume.Secret; vs != nil {
+		s, err := client.CoreV1().Secrets(namespace).Get(ctx, vs.SecretName, v1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		b, ok = s.Data[key]
+		if !ok {
+			return nil, fmt.Errorf("unable to find file key in secret: %s", key)
+		}
+	}
+	// Check for ConfigMap volume.
+	if vcm := volume.ConfigMap; vcm != nil {
+		cm, err := client.CoreV1().ConfigMaps(namespace).Get(ctx, vcm.Name, v1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		s, ok := cm.Data[key]
+		b = []byte(s)
+		if !ok {
+			return nil, fmt.Errorf("unable to find file key in configmap: %s", key)
+		}
+	}
+	return b, nil
+}
+
+func unmarshalPromConfigBytes(b []byte) (*promconfig.Config, error) {
+	var config promconfig.Config
+	// Unmarshal the data into a Prometheus config.
+	if err := yaml.Unmarshal(b, &config); err == nil {
+		return &config, nil
+		// If there were errors, try gunziping the data.
+	} else if content := http.DetectContentType(b); content == "application/x-gzip" {
+		r, err := gzip.NewReader(bytes.NewBuffer(b))
+		if err != nil {
+			return &config, err
+		}
+		ub, err := io.ReadAll(r)
```

Comment:
Should we update `ub` to a more comprehensive variable ?

Reply:
Sure - changed.